onnx/onnxifi_dummy.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * Dummy implementation of ONNX backend interface for manual test.
+ * Prints the name of the called function and backend name on each call.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+#include "onnxifi.h"
+
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI
+  ONNXIFI_SYMBOL_NAME(onnxGetBackendIDs)(
+		  onnxBackendID* backendIDs,
+		  size_t* numBackends) {
+	*numBackends = 0;
```

Comment:
numBackends/backendIDs could be nullptr also. Would be great to reflect that in the proper response flag returned, etc.
Having this will facilitate easier extension for concrete backends (No need to worry about this small things).